src/gracenote.ts

```diff
@@ -80,56 +70,40 @@ export class GraceNote extends StaveNote {
           beam.postFormat();
         }
 
-        slashBBox = this.calcBeamedNotesSlashBBox(8 * offsetScale, 8 * offsetScale, {
-          stem: 6 * offsetScale,
-          beam: 5 * offsetScale,
+        slashBBox = this.calcBeamedNotesSlashBBox(8 * scale, 8 * scale, {
+          stem: 6 * scale,
+          beam: 5 * scale,
         });
       } else {
         const stemDirection = this.getStemDirection();
         const noteHeadBounds = this.getNoteHeadBounds();
-        const noteStemHeight = stem.getHeight();
-        let x = this.getAbsoluteX();
-        let y =
-          stemDirection === Stem.DOWN ? noteHeadBounds.yTop - noteStemHeight : noteHeadBounds.yBottom - noteStemHeight;
-
-        const defaultStemExtention =
-          stemDirection === Stem.DOWN ? this.glyphProps.stemDownExtension : this.glyphProps.stemUpExtension;
-
-        let defaultOffsetY = Tables.STEM_HEIGHT;
-        defaultOffsetY -= defaultOffsetY / 2.8;
-        defaultOffsetY += defaultStemExtention;
-        y += defaultOffsetY * staveNoteScale * stemDirection;
-
-        const offsets =
-          stemDirection === Stem.UP
-            ? {
-                x1: 1,
-                y1: 0,
-                x2: 13,
-                y2: -9,
-              }
-            : {
-                x1: -4,
-                y1: 1,
-                x2: 13,
-                y2: 9,
-              };
-
-        x += offsets.x1 * offsetScale;
-        y += offsets.y1 * offsetScale;
-        slashBBox = {
-          x1: x,
-          y1: y,
-          x2: x + offsets.x2 * offsetScale,
-          y2: y + offsets.y2 * offsetScale,
-        };
+        const noteHeadWidth = this.noteHeads[0].getWidth();
+        const x = stemDirection === Stem.DOWN ? this.getAbsoluteX() : this.getAbsoluteX() + noteHeadWidth;
+        const defaultOffsetY = (Tables.STEM_HEIGHT * scale) / 2;
+        const y =
+          stemDirection === Stem.DOWN ? noteHeadBounds.yBottom + defaultOffsetY : noteHeadBounds.yTop - defaultOffsetY;
+
+        if (stemDirection === Stem.DOWN)
+          slashBBox = {
+            x1: x - noteHeadWidth,
+            y1: y - noteHeadWidth,
+            x2: x + noteHeadWidth,
+            y2: y + noteHeadWidth,
+          };
+        else
+          slashBBox = {
+            x1: x - noteHeadWidth,
+            y1: y + noteHeadWidth,
+            x2: x + noteHeadWidth,
+            y2: y - noteHeadWidth,
+          };
```

Comment:
This is slightly confusing when skimming the code. Please use extra braces { } for the if/else blocks.

Reply:
My two reasons for preferring braces:
- In the future if someone needs to add a line of code inside the if/else case, it is easier to do so.
- It is easier to read the code and visually inspect for logic errors.

Reply:
done